### PR TITLE
Various build fixes.

### PR DESCRIPTION
**Pull request by kareem-wolfssl**

### CODE REVIEW

File: test/test_hkdf.c [hunk 1]
```diff
@@ -22,6 +22,8 @@
 
 #ifdef WP_HAVE_HKDF
 
+static int test_hkdf_fail(void);
```
Comment (SparkiDev) [inline test/test_hkdf.c:25]: Put void into function declaration below.

File: test/test_pbe.c [hunk 2]
```diff
@@ -120,6 +120,10 @@ static const unsigned char pbeParamPbes2Aes256Cbc[] = {
                       0x60,0x86,0x48,0x01,0x65,0x03,0x04,0x01,0x2A,
 };
 
+static int test_pbe_sha1_des3_pbkdf1(void);
```
Comment (SparkiDev) [inline test/test_pbe.c:123]: Put void into function declarations below.

File: test/test_rand.c [hunk 3]
```diff
@@ -22,6 +22,8 @@
 
 #ifdef WP_HAVE_RANDOM
 
+static int test_random_api(void);
```
Comment (SparkiDev) [inline test/test_rand.c:25]: Put void into function declaration.

File: test/test_tls1_prf.c [hunk 4]
```diff
@@ -22,6 +22,8 @@
 
 #ifdef WP_HAVE_TLS1_PRF
 
+static int test_tls1_prf_fail(void);
```
Comment (SparkiDev) [inline test/test_tls1_prf.c:25]: Put void into function declaration instead.

File: test/unit.c [hunk 5]
```diff
@@ -26,6 +26,8 @@
 
 #include "unit.h"
 
+static void usage(void);
```
Comment (SparkiDev) [inline test/unit.c:29]: Put void into function declaration instead.

File: README.md [hunk 6]
```diff
@@ -79,6 +79,7 @@ Remove `-DWOLFSSL_PSS_LONG_SALT -DWOLFSSL_PSS_SALT_LEN_DISCOVER` and add `--enab
 ### wolfProvider
 
 ```
+git clone https://github.com/wolfSSL/wolfProvider.git
 ./autogen.sh
```
Comment (SparkiDev) [inline README.md:83]: Add line:
  cd wolfProvider

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/wp_ecc_kmgmt.c [hunk 7]
```diff
@@ -1376,7 +1376,10 @@ static int wp_ecc_gen_set_template(wp_EccGenCtx* ctx, wp_Ecc* ecc)
         }
     }
     if (ok) {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstringop-truncation"
         XSTRNCPY(ctx->curveName, name, sizeof(ctx->curveName));
```
Comment (SparkiDev) [inline src/wp_ecc_kmgmt.c:1381]: Change this to sizeof(...) - 1 and set the last character to '\0'

File: test/test_hkdf.c (hunk 1 above)
Comment (kareem-wolfssl) [reply test/test_hkdf.c:25]: I can't remove this void, it gives me the following error:
```
test/test_hkdf.c:25:1: error: function declaration isn’t a prototype [-Werror=strict-prototypes]
   25 | static int test_hkdf_fail();
      | ^~~~~~
```
I am able to compile if I have a void both in the prototype and function.  Would that work instead?

Comment (SparkiDev) [reply test/test_hkdf.c:25]: The compiler is complaining that the function prototype doesn't have void in the parameters.
If you put into the declaration then it sees it as a prototype.
Remove line and put void in to declaration below.

File: src/wp_ecc_kmgmt.c (hunk 7 above)
Comment (anhu) [reply src/wp_ecc_kmgmt.c:1381]: oddly, I did not need this.

**Comment by anhu** [comment]
I suggest closing this in favour of sparkiDev's PR.

**Comment by anhu** [comment]
@kareem-wolfssl , I suggest closing this PR. 
I think @SparkiDev 's PR already has everything from this PR.

**Pull request closed without merging**